service/task.go

```diff
@@ -640,6 +642,78 @@ func (uis *UIServer) taskLogRaw(w http.ResponseWriter, r *http.Request) {
 	uis.render.Stream(w, http.StatusOK, data, "base", "task_log.html")
 }
 
+func (uis *UIServer) taskFileRaw(w http.ResponseWriter, r *http.Request) {
+	projCtx := MustHaveProjectContext(r)
+	if projCtx.Task == nil {
+		http.Error(w, "task not found", http.StatusNotFound)
+		return
+	}
+
+	fileName := gimlet.GetVars(r)["file_name"]
+	if fileName == "" {
+		http.Error(w, "file name not specified", http.StatusBadRequest)
+		return
+	}
+
+	taskFiles, err := artifact.GetAllArtifacts([]artifact.TaskIDAndExecution{{TaskID: projCtx.Task.Id, Execution: projCtx.Task.Execution}})
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to find artifacts for task '%s'", projCtx.Task.Id))
+		return
+	}
+	taskFiles, err = artifact.StripHiddenFiles(taskFiles, true)
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrapf(err, "unable to strip hidden files for task '%s'", projCtx.Task.Id))
+		return
+	}
+	var tFile *artifact.File
+	for _, taskFile := range taskFiles {
+		if taskFile.Name == fileName {
+			tFile = &taskFile
+			break
+		}
+	}
+	if tFile == nil {
+		uis.LoggedError(w, r, http.StatusNotFound, errors.New("file not found"))
+		return
+	}
+
+	hasContentType := false
+	for _, contentType := range uis.Settings.Ui.FileStreamingContentTypes {
+		if strings.HasPrefix(tFile.ContentType, contentType) {
+			hasContentType = true
+			break
+		}
+	}
+	if !hasContentType {
+		uis.LoggedError(w, r, http.StatusBadRequest, errors.New(fmt.Sprintf("unsupported file content type '%s'", tFile.ContentType)))
+		return
+	}
+
+	response, err := http.Get(tFile.Link)
+	if err != nil {
+		uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "downloading file"))
+		return
+
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		uis.LoggedError(w, r, response.StatusCode, errors.Errorf("failed to download file with status code: %d", response.StatusCode))
```

Comment:
can we return some of the error as well or nah?

Reply:
I think we wouldn't want to. If for whatever reason the response body is super large I wouldn't want to flush it to memory just to print it out.